### PR TITLE
Make cmu_auth into a PyPi-compatible package

### PR DESCRIPTION
**Pull request by jez**
With these changes, you can install `cmu_auth` in a project simply by running

``` console
$ pip install cmu_auth
```

Commit Summary
- 78ea915
  - Recommended Python gitignore file
- 826971f
  - Rewrite README using reStructuredText for PyPi
- a99e119
  - Add command line script for quick and easy use
- 935a332
  - Add setup.py
    
    This file acts as a sort of manifest for declaring things about the
    package.
- 3273db7
  - Move core files into folder for PyPi structure

### CODE REVIEW

File: setup.py [hunk 1]
```diff
@@ -0,0 +1,15 @@
...
+setup(name='cmu_auth',
+      version='0.1.5',
+      description='Simple python methods for authenticating to Carnegie Mellon Unversity services.',
+      url='https://github.com/willcrichton/cmu_auth',
+      author='Will Crichton',
+      author_email='willcrichton@cmu.edu',
+      license='MIT',
+      packages=['cmu_auth'],
+      install_requires=[
+          'requests',
```
Comment (willcrichton) [inline setup.py]: This should be bound to a version (I use 2.0.1) for the sake of not having things break.

File: bin/cmu_auth [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+import cmu_auth
+import sys
+import getpass
+
+USAGE = '''
+usage: %s [-h|--help]
+       %s <url> [username password]'
+'''
+
+if not (len(sys.argv) == 2 or len(sys.argv) == 4) or \
```
Comment (willcrichton) [inline bin/cmu_auth:12]: Is there a motivation for not using the builtin argument parser here?

File: bin/cmu_auth [hunk 3]
```diff
@@ -0,0 +1,28 @@
+#!/Users/jake/.virtualenvs/cmuauth-test/bin/python2.7
```
Comment (willcrichton) [inline bin/cmu_auth]: This should be something more canonical like `#!/usr/bin/python27`

File: bin/cmu_auth (hunk 2 above)
Comment (jez) [inline bin/cmu_auth:12]: Laziness XD

File: bin/cmu_auth (hunk 3 above)
Comment (jez) [inline bin/cmu_auth]: Sorry, that's my bad. I had been testing it locally, made some changes, and copied it over.

File: bin/cmu_auth (hunk 2 above)
Comment (jez) [inline bin/cmu_auth:12]: Also, for the use case here, I think this implementation is simpler.

**Comment by willcrichton** [comment]
Thanks for doing this!

**Pull request merged**